Bio/SeqIO/__init__.py

```diff
@@ -869,9 +867,15 @@ def index(filename, format, alphabet=None, key_function=None):
         alphabet,
         key_function,
     )
-    return _IndexedSeqFileDict(
-        proxy_class(filename, format), key_function, repr, "SeqRecord"
-    )
+
+    try:
+        random_access_proxy = proxy_class(filename, format)
+    except TypeError:
+        raise TypeError(
+            "Need a string or path-like objects for the filename (not a handle)"
```

Comment:
This one should be object (singular) as there is only one filename.

Reply:
Oops, got a bit too eager to fix those typos. Thanks for the catch!